src/Uno.Toolkit.UI/Helpers/SystemThemeHelper.cs

```diff
@@ -35,33 +35,55 @@ public static ApplicationTheme GetCurrentOsTheme()
 		/// Get the current theme of the application.
 		/// </summary>
 		public static ApplicationTheme GetApplicationTheme()
+			=> GetRootTheme(GetWindowRoot()?.XamlRoot);
+
+		/// <summary>
+		/// Gets a <see cref="ApplicationTheme"/> from the provided XamlRoot.
+		/// </summary>
+		public static ApplicationTheme GetRootTheme(XamlRoot? root)
 		{
-			return GetWindowRoot().ActualTheme switch
+			return (root?.Content as FrameworkElement)?.ActualTheme switch
 			{
 				ElementTheme.Light => ApplicationTheme.Light,
 				ElementTheme.Dark => ApplicationTheme.Dark,
 
 				_ => GetCurrentOsTheme(),
 			};
 		}
-		
 
 		/// <summary>
 		/// Get if the application is currently in dark mode.
 		/// </summary>
-		public static bool IsAppInDarkMode() => GetApplicationTheme() == ApplicationTheme.Dark;
-		
-		public static void SetApplicationTheme(bool darkMode) => SetApplicationTheme(darkMode ? ElementTheme.Dark : ElementTheme.Light);
+		public static bool IsAppInDarkMode()
+			=> GetRootTheme(null) == ApplicationTheme.Dark;
+
+		public static bool IsRootInDarkMode(XamlRoot root)
+			=> GetRootTheme(root) == ApplicationTheme.Dark;
+
+		public static void SetApplicationTheme(bool darkMode)
+			=> SetRootTheme(GetWindowRoot()?.XamlRoot, darkMode);
+
+		/// <summary>
+		/// Sets the theme for the provided XamlRoot
+		/// </summary>
+		/// <param name="root"></param>
+		/// <param name="darkMode"></param>
+		public static void SetRootTheme(XamlRoot? root, bool darkMode)
+			=> SetApplicationTheme(root, darkMode ? ElementTheme.Dark : ElementTheme.Light);
 
-		public static void SetApplicationTheme(ElementTheme theme)
+		public static void SetApplicationTheme(XamlRoot? root, ElementTheme theme)
 		{
-			GetWindowRoot().RequestedTheme = theme;
+			if (root?.Content is FrameworkElement fe)
+			{ 
+				fe.RequestedTheme = theme;
+			}
 		}
 
-		public static void ToggleApplicationTheme() => SetApplicationTheme(darkMode: !IsAppInDarkMode());
+		public static void ToggleApplicationTheme()
+			=> SetApplicationTheme(darkMode: !IsAppInDarkMode());
 
 		private static FrameworkElement GetWindowRoot() =>
-			XamlWindow.Current.Content as FrameworkElement ??
-			throw new InvalidOperationException($"The current window content is not {(XamlWindow.Current.Content == null ? "set" : "a FrameworkElement")}.");
+			XamlWindow.Current?.Content.XamlRoot?.Content as FrameworkElement ??
+			throw new InvalidOperationException($"The current window content is not {(XamlWindow.Current?.Content == null ? "set" : "a FrameworkElement")}.");
```

Comment:
the error msg no longer made sense in some cases
maybe chance it to a static text, like:?
> "Cannot find the xaml root for the current window."